Frontend/RestaurantDashboard/src/pages/MenuList.tsx

```diff
@@ -137,26 +146,32 @@ const MenuList = () => {
                   <span className="text-sm text-gray-700">Disponível:</span>
                   <Switch
                     checked={item.available}
-                    onChange={() => toggleAvailability(idx)}
-                    className={`${
-                      item.available ? "bg-green-500" : "bg-gray-300"
-                    } relative inline-flex h-6 w-11 items-center rounded-full transition-colors`}
+                    onChange={() => toggleAvailability(item.id - 1)} // Alterado para garantir o índice correto
+                    className={`${item.available ? "bg-green-500" : "bg-gray-300"
+                      } relative inline-flex h-6 w-11 items-center rounded-full transition-colors`}
                   >
                     <span
-                      className={`${
-                        item.available ? "translate-x-6" : "translate-x-1"
-                      } inline-block h-4 w-4 transform rounded-full bg-white transition-transform`}
+                      className={`${item.available ? "translate-x-6" : "translate-x-1"
+                        } inline-block h-4 w-4 transform rounded-full bg-white transition-transform`}
                     />
                   </Switch>
                 </div>
               </div>
 
               <div className="flex flex-col items-end gap-2">
-                <button className="flex items-center gap-1 text-sm bg-blue-500 text-white px-3 py-1 rounded hover:bg-blue-600 transition">
+                <button
+                  className="flex items-center gap-1 text-sm bg-blue-500 text-white px-3 py-1 rounded hover:bg-blue-600 transition"
+                  onClick={() => navigate(`/menu/editar/${item.id}`)}
+                >
```

Comment:
Acredito que isso seja do PR de editar itens do menu. Pode remover, por favor?

```suggestion
                <button className="flex items-center gap-1 text-sm bg-blue-500 text-white px-3 py-1 rounded hover:bg-blue-600 transition">
```